src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1462,13 +1462,14 @@ void RegisterAppInterfaceRequest::CheckResponseVehicleTypeParam(
 }
 
 void RegisterAppInterfaceRequest::SendSubscribeCustomButtonNotification() {
+  SDL_LOG_AUTO_TRACE();
   using namespace smart_objects;
   SmartObject msg_params = SmartObject(SmartType_Map);
   msg_params[strings::app_id] = connection_key();
-  msg_params[strings::name] = hmi_apis::Common_ButtonName::CUSTOM_BUTTON;
+  msg_params[strings::button_name] = hmi_apis::Common_ButtonName::CUSTOM_BUTTON;
   msg_params[strings::is_suscribed] = true;
```

Comment:
```suggestion
```

To preserve existing behavior, I think we should apply `app->SubscribeButton(CUSTOM_BUTTON)` here, and then just unsubscribe if we receive a response with an error code.

Reply:
@iCollin logic has been updated as discussed internally - now SDL subscribes to CUSTOM_BUTTON along with sending request to HMI and unsubscribes if it receives a response with an error code or by timeout.
Please take a look at https://github.com/smartdevicelink/sdl_core/pull/3743/commits/ce9fc4455093d4914dfec55a6e5003da3c28fc83

Reply:
@iCollin also please note that corresponding changes in ATF scripts were done to reflect these changes:
https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2564/commits/144dfb8ce197d2c06c9e414260e50af8f2507f99